app/controllers/admin/health/patient_referrals_controller.rb

```diff
@@ -187,6 +214,27 @@ def bulk_assign_agency
       end
     end
 
+    def bulk_assign_agency_and_care_staff
+      return unless params[:assignments].present?
+
+      num_patients = 0
+      failed_count = 0
+      params.require(:assignments).each do |_, obj|
```

Comment:
Does this work?
```suggestion
      params.require(:assignments).each_value do |obj|
```
Not sure if this is a key/value situation, but that would be slightly cleaner